RETAIL-PLAN.md

```diff
@@ -0,0 +1,58 @@
+## WoW Voiceover Porting Plan
+
+### Overview
+
+The primary goal of this plan is to port the World of Warcraft addon, WoW Voiceover, from Classic WoW Vanilla to Retail WoW. The plan involves two main components:
+
+1. Crowdsourcing voice models for each race in World of Warcraft
+2. Collecting quest/gossip text data to provide the necessary information for both voice generation and correct integration into the addon
+
+### Part 1: Crowdsourcing Voice Models (ALREADY IN PROGRESS)
+
+Develop a user-friendly website to facilitate the crowdsourcing of voice models for all races in World of Warcraft. The website should include the following features:
+
+- User registration and authentication to enable contributors to sign up and submit voice samples
+- Clear instructions and guidelines on how to provide high-quality voice samples, including preferred recording setup, script examples, and ideal pronunciation
+- A submission system for users to upload their voice samples, including categorization by race, gender, and any other relevant attributes
+- An upvoting or rating system for users to evaluate the quality of submitted voice samples and provide feedback to contributors
+- A progress tracker to monitor the development of voice models for each race and gender, encouraging users to contribute to underrepresented categories
+
+### Part 2: Collecting Quest/Gossip Text Data (LOOKING FOR HELP)
+
+#### New Feature in WoW Voiceover Addon
+
+- Add a new feature to the WoW Voiceover addon that enables users to opt-in to collect quest/gossip text data during gameplay
+- Create a lightweight companion executable that runs in the background during gameplay to upload the scraped text data to a central server
+- The data should include:
+
+| Column         | Description                                                |
+|----------------|------------------------------------------------------------|
+| source         | Type of interaction ('accept', 'progress', 'complete', or 'gossip') |
+| quest          | Quest ID or empty string if it's a gossip interaction      |
+| text           | text content of the interaction |
+| race           | Race of the NPC involved in the interaction                 |
+| sex            | Gender of the NPC involved in the interaction               |
+| npc_name           | Name of the NPC involved in the interaction                 |
+| npc_type           | Type of the NPC involved ('creature', 'gameobject', or 'item') |
+| npc_id             | Creature/gameobject/item ID of the NPC involved in the interaction |
+| player_race    | Race of the player involved in the interaction              |
+| player_gender  | Gender of the player involved in the interaction            |
+| player_name  | Name of the player involved in the interaction            |
+
+- Race and Sex will probably not be possible to collect directly from the game addon for recent/current expansions.
```

Comment:
Race doesn't seem possible to obtain using in-game API, because it carries no gameplay value for non-playable characters.

Gender is possible to obtain using in-game API, but only if blizzard has bothered assigning it correctly. The reality is that most creatures will have a "neutral" gender, even for blatantly male/female-looking models.

Obtaining both for playable race models (i.e. if the creature is using the same model as players do when creating a character of that race) is trivial using out-of-game tools (creature cache -> displayid -> CreatureDisplayInfo.db2 -> CreatureDisplayInfoExtra.db2), but for non-playable models that info is unavailable and it will have to be guesswork, as outlined in the "NPC Race and Sex Matching" section.

Reply:
done